addons/l10n_ch/models/res_bank.py

```diff
@@ -81,60 +47,18 @@ class ResPartnerBank(models.Model):
                                        "still use the main IBAN in the Account Number while you will see the "
                                        "QR-IBAN for the barcode.  ")
 
-    # fields to configure ISR payment slip generation
-    l10n_ch_isr_subscription_chf = fields.Char(string='CHF ISR Subscription Number', help='The subscription number provided by the bank or Postfinance to identify the bank, used to generate ISR in CHF. eg. 01-162-8')
-    l10n_ch_isr_subscription_eur = fields.Char(string='EUR ISR Subscription Number', help='The subscription number provided by the bank or Postfinance to identify the bank, used to generate ISR in EUR. eg. 03-162-5')
-    l10n_ch_show_subscription = fields.Boolean(compute='_compute_l10n_ch_show_subscription', default=lambda self: self.env.company.account_fiscal_country_id.code == 'CH')
-
-    def _is_isr_issuer(self):
-        return (_is_l10n_ch_isr_issuer(self.l10n_ch_postal, 'CHF')
-                or _is_l10n_ch_isr_issuer(self.l10n_ch_postal, 'EUR'))
-
-    @api.constrains("l10n_ch_postal", "partner_id")
-    def _check_postal_num(self):
-        """Validate postal number format"""
-        for rec in self:
-            if rec.l10n_ch_postal and not _is_l10n_ch_postal(rec.l10n_ch_postal):
-                # l10n_ch_postal is used for the purpose of Client Number on your own accounts, so don't do the check there
-                if rec.partner_id and not rec.partner_id.ref_company_ids:
-                    raise ValidationError(
-                        _("The postal number {} is not valid.\n"
-                          "It must be a valid postal number format. eg. 10-8060-7").format(rec.l10n_ch_postal))
-        return True
-
-    @api.constrains("l10n_ch_isr_subscription_chf", "l10n_ch_isr_subscription_eur")
-    def _check_subscription_num(self):
-        """Validate ISR subscription number format
-        Subscription number can only starts with 01 or 03
-        """
-        for rec in self:
-            for currency in ["CHF", "EUR"]:
-                subscrip = rec.l10n_ch_isr_subscription_chf if currency == "CHF" else rec.l10n_ch_isr_subscription_eur
-                if subscrip and not _is_l10n_ch_isr_issuer(subscrip, currency):
-                    example = "01-162-8" if currency == "CHF" else "03-162-5"
-                    raise ValidationError(
-                        _("The ISR subcription {} for {} number is not valid.\n"
-                          "It must starts with {} and we a valid postal number format. eg. {}"
-                          ).format(subscrip, currency, ISR_SUBSCRIPTION_CODE[currency], example))
-        return True
+    # fields to configure payment slip generation
+    l10n_ch_bank_is_ch_or_li = fields.Boolean(compute='_compute_l10n_ch_bank_is_ch_or_li', tracking=True)
```

Comment:
I don't really like the new name of this field: it explains how the computation is done, not the use of the field, and it should rather do that.